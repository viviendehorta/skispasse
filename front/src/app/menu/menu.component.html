<div class="sk-menu">
    <v-menu-logo></v-menu-logo>
</div>
