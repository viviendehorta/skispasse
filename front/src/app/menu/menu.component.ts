import {Component} from '@angular/core'

@Component({
    selector: "s-menu",
    templateUrl: "./menu.component.html"
})
export class MenuComponent {

    constructor() {
    }
}
