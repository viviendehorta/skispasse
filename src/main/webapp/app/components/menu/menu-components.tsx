import React from 'react';

import {DropdownMenu, DropdownToggle, UncontrolledDropdown} from 'reactstrap';
import {FontAwesomeIcon} from '@fortawesome/react-fontawesome';


export const BrandIcon = props => (
  <div {...props} className="brand-icon">
    <img src="content/images/logo-jhipster.png" alt="Logo"/>
  </div>
);

export const NavDropdown = props => (
  <UncontrolledDropdown nav inNavbar id={props.id}>
    <DropdownToggle nav caret className="menu-nav-item d-flex align-items-center">
      <FontAwesomeIcon icon={props.icon}/>
      <span>{props.name}</span>
    </DropdownToggle>
    <DropdownMenu right style={props.style}>
      {props.children}
    </DropdownMenu>
  </UncontrolledDropdown>
);
