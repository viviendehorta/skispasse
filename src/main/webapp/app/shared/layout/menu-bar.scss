$header-color: #fff;
$header-color-secondary: #bbb;
$header-color-hover: darken($header-color, 20%);

$header-bar-without-toggle-btn-width: 14rem;
$header-toggle-btn-width: 3rem;
$header-toggle-btn-margin: 1rem;
$header-bar-width: $header-bar-without-toggle-btn-width + $header-toggle-btn-margin + $header-toggle-btn-width;

.app-header {
  position: absolute;

  .loading-bar {
    height: 3px;
    background-color: #009cd8;
    position: absolute;
    top: 0px;
    z-index: 1031;
  }

  .header-bar {
    height: 100vh;
    min-width: $header-bar-width;
    width: $header-bar-width;
    background-color: transparent;
    transition: all 0.4s;

    .header-bar-without-toggle-btn {
      min-width: $header-bar-without-toggle-btn-width;
      width: $header-bar-without-toggle-btn-width;
      height: 100vh;
      box-shadow: 3px 3px 10px rgba(0, 0, 0, 0.1);
      background-color: white;

      .brand-title {
        font-size: 24px;
        color: $header-color;
        padding-top: 2.5rem;
        padding-bottom: 2.5rem;

        &:hover {
          color: $header-color-hover;
          text-decoration: none;
        }
      }

      .navbar-version {
        font-size: 10px;
        color: $header-color-secondary;
        padding: 0 0 0 10px;
      }

      .header-group-label {
        font-size: 80%;
        font-weight: 700;
        text-transform: uppercase;
        margin-bottom: 0;
        padding: 1.5rem 1rem;
      }

      .header-nav-item {
        color: #2c3e50;
        font-style: italic;
        padding-top: 0.8rem;
        padding-bottom: 0.8rem;
      }

      .header-nav-item:hover,
      .header-nav-item:focus {
        color: #576263;
      }
    }

    .header-collapse-btn {
      position: absolute;
      top: 0;
      right: 0;
      margin-top: $header-toggle-btn-margin;
      margin-left: $header-toggle-btn-margin;
      text: uppercase;
      font-weight: bold;
      width: $header-toggle-btn-width;
      border-radius: 0.5rem;
    }

    .header-collapse-btn:visited,
    .header-collapse-btn:focus {
      box-shadow: none;
    }
  }
}

/* for toggle behavior */

#header-bar.active {
  margin-left: -$header-bar-without-toggle-btn-width;
}

@media (max-width: 768px) {
  #header-bar {
    margin-left: -$header-bar-without-toggle-btn-width;
  }
  #header-bar.active {
    margin-left: 0;
  }
}
