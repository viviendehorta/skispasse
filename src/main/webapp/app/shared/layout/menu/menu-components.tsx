import React from 'react';

import {DropdownMenu, DropdownToggle, NavItem, NavLink, UncontrolledDropdown} from 'reactstrap';
import {FontAwesomeIcon} from '@fortawesome/react-fontawesome';
import {Translate} from 'react-jhipster';
import {NavLink as Link} from 'react-router-dom';

import appConfig from 'app/config/constants';


export const BrandIcon = props => (
  <div {...props} className="brand-icon">
    <img src="content/images/logo-jhipster.png" alt="Logo"/>
  </div>
);

export const Brand = props => (
  <NavLink tag={Link} to="/" className="brand-title menu-nav-item">
    <span>
      <Translate contentKey="global.title">Skispasse</Translate>
    </span>
    <span className="navbar-version">{appConfig.VERSION}</span>
  </NavLink>
);

export const Home = props => (
  <NavItem>
    <NavLink tag={Link} to="/" className="menu-nav-item">
      <span>
        <FontAwesomeIcon icon="home"></FontAwesomeIcon>
      </span>
      <Translate contentKey="global.menu.home">Home</Translate>
    </NavLink>
  </NavItem>
);

export const MapMenu = props => (
  <NavItem>
    <NavLink tag={Link} to="/worldmap" className="menu-nav-item">
      <FontAwesomeIcon icon="map"/>
      <span>
        <Translate contentKey="global.menu.worldmap">Map</Translate>
      </span>
    </NavLink>
  </NavItem>
);

export const NavDropdown = props => (
  <UncontrolledDropdown nav inNavbar id={props.id}>
    <DropdownToggle nav caret className="menu-nav-item d-flex align-items-center">
      <FontAwesomeIcon icon={props.icon}/>
      <span>{props.name}</span>
    </DropdownToggle>
    <DropdownMenu right style={props.style}>
      {props.children}
    </DropdownMenu>
  </UncontrolledDropdown>
);
