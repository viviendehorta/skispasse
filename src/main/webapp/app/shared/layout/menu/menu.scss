$menu-color: #fff;
$menu-color-secondary: #bbb;
$menu-color-hover: darken($menu-color, 20%);

$menu-without-toggle-btn-width: 14rem;
$menu-toggle-btn-width: 3rem;
$menu-toggle-btn-margin: 1rem;
$menu-width: $menu-without-toggle-btn-width + $menu-toggle-btn-margin + $menu-toggle-btn-width;

.app-menu {
  position: absolute;

  .loading-bar {
    height: 3px;
    background-color: #009cd8;
    position: absolute;
    top: 0px;
    z-index: 1031;
  }

  .menu {
    height: 100vh;
    min-width: $menu-width;
    width: $menu-width;
    background-color: transparent;
    transition: all 0.4s;

    .menu-without-toggle-btn {
      min-width: $menu-without-toggle-btn-width;
      width: $menu-without-toggle-btn-width;
      height: 100vh;
      box-shadow: 3px 3px 10px rgba(0, 0, 0, 0.1);
      background-color: white;

      .brand-title {
        font-size: 24px;
        color: $menu-color;
        padding-top: 2.5rem;
        padding-bottom: 2.5rem;

        &:hover {
          color: $menu-color-hover;
          text-decoration: none;
        }
      }

      .navbar-version {
        font-size: 10px;
        color: $menu-color-secondary;
        padding: 0 0 0 10px;
      }

      .menu-group-label {
        font-size: 80%;
        font-weight: 700;
        text-transform: uppercase;
        margin-bottom: 0;
        padding: 1.5rem 1rem;
      }

      .menu-nav-item {
        color: #2c3e50;
        font-style: italic;
        padding-top: 0.8rem;
        padding-bottom: 0.8rem;
      }

      .menu-nav-item:hover,
      .menu-nav-item:focus {
        color: #576263;
      }
    }

    .menu-collapse-btn {
      position: absolute;
      top: 0;
      right: 0;
      margin-top: $menu-toggle-btn-margin;
      margin-left: $menu-toggle-btn-margin;
      text: uppercase;
      font-weight: bold;
      width: $menu-toggle-btn-width;
      border-radius: 0.5rem;
    }

    .menu-collapse-btn:visited,
    .menu-collapse-btn:focus {
      box-shadow: none;
    }
  }
}

/* for toggle behavior */

#menu.active {
  margin-left: -$menu-without-toggle-btn-width;
}

@media (max-width: 768px) {
  #menu {
    margin-left: -$menu-without-toggle-btn-width;
  }
  #menu.active {
    margin-left: 0;
  }
}
