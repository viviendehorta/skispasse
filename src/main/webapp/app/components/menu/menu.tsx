import './menu.scss';

import React, {useState} from 'react';
import {Storage} from 'react-jhipster';
import {Button} from 'reactstrap';
import {FontAwesomeIcon} from '@fortawesome/react-fontawesome';

import {Brand, Home, MapMenu} from './menu-components';
import {AccountMenu, AdminMenu, LocaleMenu} from './';

export interface IMenuProps {
  isAuthenticated: boolean;
  isAdmin: boolean;
  isInProduction: boolean;
  isSwaggerEnabled: boolean;
  currentLocale: string;
  onLocaleChange: Function;
}

const Menu = (props: IMenuProps) => {

  const handleLocaleChange = event => {
    const langKey = event.target.value;
    Storage.session.set('locale', langKey);
    props.onLocaleChange(langKey);
  };

  const [isCollapsed, setIsCollapsed] = useState(false);

  const toggleMenu = function () {
    setIsCollapsed(!isCollapsed);
  };

  return (
    <div className="app-menu">

      <div className={isCollapsed ? "menu collapsed" : "menu"} id="menu">

        <div className="menu-without-toggle-btn">

          <Brand/>

          <ul className="nav flex-column bg-white mb-0">

            <Home/>

            <MapMenu/>

            <LocaleMenu currentLocale={props.currentLocale} onClick={handleLocaleChange}/>

            <AccountMenu isAuthenticated={props.isAuthenticated}/>

            {props.isAuthenticated && props.isAdmin && <AdminMenu showSwagger={props.isSwaggerEnabled}/>}

          </ul>

        </div>

        <Button id="menuCollapseBtn" className="menu-collapse-btn" onClick={toggleMenu}>
          <FontAwesomeIcon icon="bars" className="menu-collapse-icon"/>
        </Button>

      </div>
    </div>
  );
};

export default Menu;
