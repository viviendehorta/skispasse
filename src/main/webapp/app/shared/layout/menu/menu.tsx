import './menu.scss';

import React, {useState} from 'react';
import {Storage} from 'react-jhipster';
import {Button} from 'reactstrap';
import LoadingBar from 'react-redux-loading-bar';
import {FontAwesomeIcon} from '@fortawesome/react-fontawesome';

import {Brand, Home, MapMenu} from './menu-components';
import {AccountMenu, AdminMenu, EntitiesMenu, LocaleMenu} from './';

export interface IMenuProps {
  isAuthenticated: boolean;
  isAdmin: boolean;
  isInProduction: boolean;
  isSwaggerEnabled: boolean;
  currentLocale: string;
  onLocaleChange: Function;
}

const Menu = (props: IMenuProps) => {

  const handleLocaleChange = event => {
    const langKey = event.target.value;
    Storage.session.set('locale', langKey);
    props.onLocaleChange(langKey);
  };

  const [isCollapsed, setIsCollapsed] = useState(false);

  const toggleMenu = function () {
    setIsCollapsed(!isCollapsed);
  };

  return (
    <div className="app-menu">
      <LoadingBar className="loading-bar"/>

      <div className={isCollapsed ? "menu collapsed" : "menu"} id="menu">

        <div className="menu-without-toggle-btn">

          <Brand/>

          <ul className="nav flex-column bg-white mb-0">

            <Home/>

            <MapMenu/>

            <LocaleMenu currentLocale={props.currentLocale} onClick={handleLocaleChange}/>

            <AccountMenu isAuthenticated={props.isAuthenticated}/>

            {props.isAuthenticated && <EntitiesMenu/>}

            {props.isAuthenticated && props.isAdmin && <AdminMenu showSwagger={props.isSwaggerEnabled}/>}

          </ul>

        </div>

        <Button id="menuCollapseBtn" className="menu-collapse-btn" onClick={toggleMenu}>
          <FontAwesomeIcon icon="bars" className="menu-collapse-icon"/>
        </Button>

      </div>
    </div>
  );
};

export default Menu;
