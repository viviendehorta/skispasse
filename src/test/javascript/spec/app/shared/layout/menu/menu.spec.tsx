import React from 'react';
import { shallow } from 'enzyme';

import sinon from 'sinon';

import LoadingBar from 'react-redux-loading-bar';

import { Home, Brand } from 'app/shared/layout/menu/menu-components';
import { AdminMenu, AccountMenu, LocaleMenu } from 'app/shared/layout/menu';
import Menu from 'app/shared/layout/menu/menu';

describe('Menu', () => {
  let mountedWrapper;

  const localeSpy = sinon.spy();

  const devProps = {
    isAuthenticated: true,
    isAdmin: true,
    currentLocale: 'en',
    onLocaleChange: localeSpy,
    isInProduction: false,
    isSwaggerEnabled: true
  };
  const prodProps = {
    ...devProps,
    isInProduction: true,
    isSwaggerEnabled: false
  };
  const userProps = {
    ...prodProps,
    isAdmin: false
  };
  const guestProps = {
    ...prodProps,
    isAdmin: false,
    isAuthenticated: false
  };

  const wrapper = (props = devProps) => {
    if (!mountedWrapper) {
      mountedWrapper = shallow(<Menu {...props} />);
    }
    return mountedWrapper;
  };

  beforeEach(() => {
    mountedWrapper = undefined;
  });

  // All tests will go here
  it('Renders a Menu component in dev profile with LoadingBar and Nav.', () => {
    const component = wrapper();
    // the created snapshot must be committed to source control
    expect(component).toMatchSnapshot();
    expect(component.find(LoadingBar).length).toEqual(1);
    const menu = component.find('.menu');
    expect(menu.length).toEqual(1);
    expect(menu.find(Brand).length).toEqual(1);
    const nav = component.find('.nav');
    expect(nav.length).toEqual(1);
    expect(nav.find(Home).length).toEqual(1);
    expect(nav.find(AdminMenu).length).toEqual(1);
    expect(nav.find(LocaleMenu).length).toEqual(1);

    expect(nav.find(AccountMenu).length).toEqual(1);
  });

  it('Renders a Menu component in prod profile with LoadingBar and Nav.', () => {
    const component = wrapper(prodProps);
    // the created snapshot must be committed to source control
    expect(component).toMatchSnapshot();
    const menu = component.find('.menu');
    expect(menu.length).toEqual(1);
    expect(menu.find(Brand).length).toEqual(1);
    const nav = component.find('.nav');
    expect(nav.length).toEqual(1);
    expect(nav.find(Home).length).toEqual(1);
    expect(nav.find(AdminMenu).length).toEqual(1);
    expect(nav.find(LocaleMenu).length).toEqual(1);

    expect(nav.find(AccountMenu).length).toEqual(1);
  });

  it('Renders a Menu component in prod profile with logged in User', () => {
    const nav = wrapper(userProps).find('.nav');
    expect(nav.find(AdminMenu).length).toEqual(0);
    const account = nav.find(AccountMenu);
    expect(account.first().props().isAuthenticated).toEqual(true);
  });

  it('Renders a Menu component in prod profile with no logged in User', () => {
    const nav = wrapper(guestProps).find('.nav');
    expect(nav.find(AdminMenu).length).toEqual(0);
    const account = nav.find(AccountMenu);
    expect(account.length).toEqual(1);
    expect(account.first().props().isAuthenticated).toEqual(false);
  });

  it('Renders an uncollapsed Menu component', () => {
    const nav = wrapper(devProps).find('.nav');
    expect(nav.hasClass('collapsed')).toBeFalsy();
  });
});
