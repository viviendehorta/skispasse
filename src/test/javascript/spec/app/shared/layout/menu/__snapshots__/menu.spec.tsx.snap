// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Menu Renders a Menu component in dev profile with LoadingBar and Nav. 1`] = `
<div
  className="app-menu"
>
  <Connect(LoadingBar)
    className="loading-bar"
  />
  <div
    className="menu"
    id="menu"
  >
    <div
      className="menu-without-toggle-btn"
    >
      <Component />
      <ul
        className="nav flex-column bg-white mb-0"
      >
        <Component />
        <Component />
        <Component
          currentLocale="en"
          onClick={[Function]}
        />
        <Component
          isAuthenticated={true}
        />
        <Component
          showSwagger={true}
        />
      </ul>
    </div>
    <Button
      className="menu-collapse-btn"
      color="secondary"
      id="menuCollapseBtn"
      onClick={[Function]}
      tag="button"
    >
      <FontAwesomeIcon
        border={false}
        className="menu-collapse-icon"
        fixedWidth={false}
        flip={null}
        icon="bars"
        inverse={false}
        listItem={false}
        mask={null}
        pull={null}
        pulse={false}
        rotation={null}
        size={null}
        spin={false}
        symbol={false}
        title=""
        transform={null}
      />
    </Button>
  </div>
</div>
`;

exports[`Menu Renders a Menu component in prod profile with LoadingBar and Nav. 1`] = `
<div
  className="app-menu"
>
  <Connect(LoadingBar)
    className="loading-bar"
  />
  <div
    className="menu"
    id="menu"
  >
    <div
      className="menu-without-toggle-btn"
    >
      <Component />
      <ul
        className="nav flex-column bg-white mb-0"
      >
        <Component />
        <Component />
        <Component
          currentLocale="en"
          onClick={[Function]}
        />
        <Component
          isAuthenticated={true}
        />
        <Component
          showSwagger={false}
        />
      </ul>
    </div>
    <Button
      className="menu-collapse-btn"
      color="secondary"
      id="menuCollapseBtn"
      onClick={[Function]}
      tag="button"
    >
      <FontAwesomeIcon
        border={false}
        className="menu-collapse-icon"
        fixedWidth={false}
        flip={null}
        icon="bars"
        inverse={false}
        listItem={false}
        mask={null}
        pull={null}
        pulse={false}
        rotation={null}
        size={null}
        spin={false}
        symbol={false}
        title=""
        transform={null}
      />
    </Button>
  </div>
</div>
`;
